### PR TITLE
DEP: Enforce numpy keyword deprecation in read_json

### PR DESCRIPTION
**Pull request by phofl**
#30636

### CODE REVIEW

**Comment by WillAyd** [comment]
Can you also remove the relevant code from JSONToObj.c?

**Comment by WillAyd** [comment]
Some pointers that may be helpful:

1. Should be able to remove the dtype struct member:
https://github.com/pandas-dev/pandas/blob/8b503a8ca06c6370fc3fbd972fc6b0a621df9531/pandas/_libs/src/ujson/python/JSONtoObj.c#L54

2. Can remove from kwargs:
https://github.com/pandas-dev/pandas/blob/8b503a8ca06c6370fc3fbd972fc6b0a621df9531/pandas/_libs/src/ujson/python/JSONtoObj.c#L497

3. Can remove local variable reference:
https://github.com/pandas-dev/pandas/blob/8b503a8ca06c6370fc3fbd972fc6b0a621df9531/pandas/_libs/src/ujson/python/JSONtoObj.c#L508

And any references of that variable you find

**Comment by phofl** [comment]
Not sure if I got everything.

I don't think we can remove the dtype, it is also used for dtype inference if nothing is passed

File: pandas/tests/io/json/test_ujson.py [hunk 1]
```diff
@@ -218,10 +198,7 @@ def test_encode_array_of_nested_arrays(self):
         assert nested_input == json.loads(output)
         assert nested_input == ujson.decode(output)
 
-        nested_input = np.array(nested_input)
-        tm.assert_numpy_array_equal(
-            nested_input, ujson.decode(output, numpy=True, dtype=nested_input.dtype)
-        )
+        assert nested_input == ujson.decode(output)
```
Comment (mroeschke) [inline pandas/tests/io/json/test_ujson.py:201]: Looks like this is already asserted above.

File: pandas/tests/io/json/test_ujson.py [hunk 2]
```diff
@@ -230,9 +207,7 @@ def test_encode_array_of_doubles(self):
         assert doubles_input == json.loads(output)
         assert doubles_input == ujson.decode(output)
 
-        tm.assert_numpy_array_equal(
-            np.array(doubles_input), ujson.decode(output, numpy=True)
-        )
+        assert doubles_input == ujson.decode(output)
```
Comment (mroeschke) [inline pandas/tests/io/json/test_ujson.py:210]: Same (and a few more below)

File: pandas/tests/io/json/test_ujson.py (hunk 1 above)
Comment (phofl) [reply pandas/tests/io/json/test_ujson.py]: Hm weird, thx. Removed

**Comment by WillAyd** [comment]
Nice cleanup - lgtm after conflict fixup

**Pull request merged**